### PR TITLE
#162948898 upgrade django from 1.10 to 2.1.4

### PR DESCRIPTION
**Pull request by danuluma**
#### What does this PR do?
Upgrade Django from version 1.10 to 2.1.4

#### Description of Task(s) to be completed?
* Upgrade Django from v1.10 to v2.1.4
* Upgrade all the other 3rd party packages used in the project to their latest versions
* Remove imports that were available in Django 1.10(`django.conf.urls`) but deprecated in Django 2.1.4, and use updated ones instead(`django.urls`)
* Temporarily comment out JWT Authentication from the list of default authentication classes since it hasn't been implemented yet
* Updated all references to Django v1.1 in the docstrings and comments to Django v2.1.4

#### How should this be manually tested?
* Pull the branch `git fetch origin ch-django-upgrade-162948898:ch-django-upgrade-162948898`
* Checkout to the branch `git checkout ch-django-upgrade-162948898`
* Install requirements `pip3 install -r requirements.txt`
* Check the versions of the installed packages by running `pip3 freeze` in your terminal

#### Any background context you want to provide?
* This project uses the latest version of python3. Please download it from [here](https://www.python.org/downloads/)
* You may may as well use pip instead of pip3 if you have either only python3 installed on your computer or you have it set as the default version of python

#### What are the relevant pivotal tracker stories?
[#162948898](https://www.pivotaltracker.com/story/show/162948898)

### CODE REVIEW

**Review by tesh254 — APPROVED** [review]

**Review by Ruiru11 — APPROVED** [review]

**Review by Deekerubo — APPROVED** [review]

**Review by boltonotieno — APPROVED** [review]

**Review by Ruiru11 — COMMENTED** [review]
your commit message does not meet the standards set out in the Andela engineering playbook review and fix that please

**Comment by danuluma** [comment]
Okay. I'm updating it

**Review by Ruiru11 — APPROVED** [review]
awesome

**Comment by frankip** [comment]
@danuluma In you PR description in the How should this be manually tested 
The best way to describe it is to 
```
- pull the branch `git fetch origin branch-name:branch-name`
- checkout to the `branch git checkout branch_name`
- Install requirements `pip install -r requirements.txt`
........
```
something similar to that

File: authors/settings.py [hunk 1]
```diff
@@ -138,6 +138,6 @@
     'NON_FIELD_ERRORS_KEY': 'error',
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
+        # 'authors.apps.authentication.backends.JWTAuthentication',
```
Comment (frankip) [inline authors/settings.py:141]: @danuluma why is this line commented out?

**Comment by danuluma** [comment]
> @danuluma In you PR description in the How should this be manually tested
> The best way to describe it is to
> 
> ```
> - pull the branch `git fetch origin branch-name:branch-name`
> - checkout to the `branch git checkout branch_name`
> - Install requirements `pip install -r requirements.txt`
> ........
> ```
> something similar to that

Ok thanks. Let me update it accordingly

File: authors/settings.py (hunk 1 above)
Comment (danuluma) [reply authors/settings.py:141]: JWT authentication hasn't been implemented yet. Having it as one of the authentication classes causes the app to throw an error when one tries to run it.

Comment (danuluma) [reply authors/settings.py:141]: Should I uncomment it?

Comment (boltonotieno) [reply authors/settings.py:141]: Travis also throws that error.
image

Comment (frankip) [reply authors/settings.py:141]: Okay you can leave it like that

**Review by frankip — APPROVED** [review]

**Pull request merged**